File: src/Participant.java
// Учасник проєкту: базовий абстрактний клас
public abstract class Participant {
    private int id;
    private String name;

    // Конструктор для ініціалізації учасника
    public Participant(int id, String name) {
        this.id = id;
    
        this.id = id;
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    // Абстрактний метод для ролі учасника
    public abstract String getRole();
}

File: construction/python/Participant.py
from abc import ABC, abstractmethod

class Participant(ABC):
    def __init__(self, id: int, name: str):
        self._id = id
        self._name = name

    def get_id(self) -> int:
        return self._id

    def get_name(self) -> str:
        return self._name

    # Абстрактний метод для ролі учасника
    @abstractmethod
    def get_role(self) -> str:
        pass